Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp

```diff
@@ -152,51 +161,141 @@ ActsExamples::ProcessCode ActsExamples::TrackFindingAlgorithm::execute(
 
   unsigned int nSeed = 0;
 
-  for (std::size_t iseed = 0; iseed < initialParameters.size(); ++iseed) {
+  for (std::size_t iSeed = 0; iSeed < initialParameters.size(); ++iSeed) {
     // Clear trackContainerTemp and trackStateContainerTemp
     tracksTemp.clear();
 
-    auto result =
-        (*m_cfg.findTracks)(initialParameters.at(iseed), options, tracksTemp);
+    const Acts::BoundTrackParameters& firstInitialParameters =
+        initialParameters.at(iSeed);
+
+    auto firstResult =
+        (*m_cfg.findTracks)(firstInitialParameters, firstOptions, tracksTemp);
     m_nTotalSeeds++;
     nSeed++;
 
-    if (!result.ok()) {
+    if (!firstResult.ok()) {
       m_nFailedSeeds++;
-      ACTS_WARNING("Track finding failed for seed " << iseed << " with error"
-                                                    << result.error());
+      ACTS_WARNING("Track finding failed for seed " << iSeed << " with error"
+                                                    << firstResult.error());
       continue;
     }
 
-    auto& tracksForSeed = result.value();
-    for (auto& track : tracksForSeed) {
-      auto smoothingResult = Acts::smoothTrack(ctx.geoContext, track, logger());
-      if (!smoothingResult.ok()) {
+    auto& firstTracksForSeed = firstResult.value();
+    for (auto& firstTrack : firstTracksForSeed) {
+      // TODO select track
```

Comment:
why might we want to select the track at this stage?

Reply:
This could be an opportunity to throw tracks out already. Either before smoothing or after or both. Cuts on the impact parameters are not possible but we could cut on momentum at innermost hit or number of measurements or holes.

But this makes only sense if we can cut on more or "better" information compared to the branch aborter

Reply:
yeah. I think this is best handled in the branch aborter which can stop once it has enough information to make the cut.